app/src/main/java/com/example/plantsapp/presentation/ui/authentication/AuthViewModel.kt

```diff
@@ -11,36 +11,42 @@ import com.example.plantsapp.presentation.core.Event
 import dagger.hilt.android.lifecycle.HiltViewModel
 import kotlinx.coroutines.launch
 import timber.log.Timber
+import java.lang.Exception
 import javax.inject.Inject
 
 @HiltViewModel
 class AuthViewModel @Inject constructor(
     private val authUseCase: AuthUseCase,
 ) : ViewModel() {
 
-    sealed class AuthResult {
-        object NavigateToTasks : AuthResult()
-        data class AuthError(@StringRes val errorId: Int) : AuthResult()
+    sealed class AuthState {
+        object Loading : AuthState()
+        object NavigateToTasks : AuthState()
+        data class AuthError(@StringRes val errorId: Int) : AuthState()
     }
 
-    private val _authResult: MutableLiveData<Event<AuthResult>> = MutableLiveData()
-    val authResult: LiveData<Event<AuthResult>> get() = _authResult
+    private val _authState: MutableLiveData<Event<AuthState>> = MutableLiveData()
+    val authState: LiveData<Event<AuthState>> get() = _authState
 
+    @Suppress("TooGenericExceptionCaught")
     fun onSignInResult(token: String?) {
         viewModelScope.launch {
             if (token == null) {
-                _authResult.value = Event(AuthResult.AuthError(R.string.error_unable_to_sign_in))
+                _authState.value =
+                    Event(AuthState.AuthError(R.string.error_unable_to_sign_in))
                 return@launch
             }
 
             val event = try {
+                _authState.value = Event(AuthState.Loading)
```

Comment:
There is one issue - AuthState.Loading is state, but you need to process it like state, not like event

To check what happens if you process this like an event, try add delay(5000) after this line, and while delay is running , rotate your phone